### PR TITLE
feat(automation): add gauge-valve radial position gauge

### LINKED ISSUE #1169: Add obc-gauge-valve: radial valve position gauge (6.1 Gauge-valves)
## What

`obc-gauge-valve` — a circular analog gauge showing valve position, from the OpenBridge 6.1 *Gauge-valves* Figma masters. Two variants:

- **Two-way** — two 90° port tracks
- **Three-way** — three 60° tracks plus a bottom port

Each track carries a fill bar for its port's flow value, with a cap pill at the end.

## Why

The library has no radial valve-position gauge. Existing automation valve components (`obc-analog-valve`, `obc-digital-valve`) are P&ID symbols showing on/off or a linear position, not a dial reading per-port flow. Consumers building a valve detail view have to compose it themselves.

## Scope

| Axis | Values |
| --- | --- |
| Size | `compact` (label stack: setpoint row, per-port % values, tag line) and `large` (0–100 scale, setpoint marker, instrument readout) |
| `priority` | `regular` / `enhanced` / `medium` / `off` |
| `barStyle` | `tint` / `fill` |

- Pure-function geometry module so the track/cap/tick math is unit-testable independently of the render.
- Sizing goes through the shared `computeRadialFrame()` (contain-fit, label reserve, `faceDiameter`), the watch coordinate space, `roundedArch()`, the shared setpoint layer and `renderInstrumentReadout`.

## Design reference

Figma OpenBridge 6.1 *Gauge-valves*. Geometry verified against node metadata and pixel measurement: track annulus r116–164, cap pills 8px flush with the track, ticks 168→176/188 touching the r188 outline, face ending at the outline.

## Known follow-ups

`obc-gauge-valve` draws its face bespoke instead of composing `obc-watch`, because its fixed-angle port tracks sit at a band radius `watch.ts` cannot currently render and its scale spans a fixed 60° top arc. This is recorded as the sanctioned **partial-reuse exception** in `docs/agents/watch-radial-instruments.md`, not a precedent. The mapping onto existing watch primitives is close enough that a future rebuild scopes as "add detached band-sector tracks + a corner-radius knob to `watch.ts`":

- track annulus 112–160 = `WatchCircleType.double`'s secondary-ring band (`RING2 160` / `RING3 112`), and `WatchBarArea`'s default radii
- fill bars = `barAreas` at default radii
- cap pills (8 wide, rounded, spanning 112–160) = the `WatchNeedle` pill shape
- tick radii reuse watch tick geometry numerically (`164 = 328/2`, `172 = 344/2`, `184 = 368/2`) but are re-declared as local literals, so nothing keeps them in sync with `tickmark.ts`

Genuine gaps in `watch.ts` today: per-sector band **tracks** detached from the outer ring with `roundRadius: 5` corner fillets, a fixed 60° top scale with labels at fixed SVG r 198, and the center icon slot.

Other open items:

- Setpoint confirm animation is unsupported here. When added, the `cssSafeAngle()` accumulated-angle + CSS-transition pattern from `watch.ts` / `docs/agents/setpoint.md` is the intended shared mechanism.
- The compact label stack hand-rolls padded values (`renderPaddedValue()` + local CSS). #1085 extends `obc-automation-button-readout-stack` with the same ghost-zero padding — once both land, the valve's stack could become a consumer of the shared stack instead of a second implementation.

## Implementation

PR #1100.

### PR DESCRIPTION
**Pull request by ludvigovrevik**
## Summary
- Adds `obc-gauge-valve`, a circular analog valve position gauge (OpenBridge 6.1 *Gauge-valves* Figma masters)
- Two-way (90° tracks) and three-way (60° tracks + bottom port) variants
- Compact variant with a label stack (setpoint row, per-port % values, tag line); `large` variant with 0–100 scale, setpoint marker and instrument readout
- `priority` (regular/enhanced/medium/off) and `barStyle` (tint/fill) color axes
- Geometry verified against Figma node metadata and pixel measurement: track annulus r116–164, cap pills 8px flush with the track, ticks 168→176/188 touching the r188 outline, face ending at the outline
- Pure-function geometry module with 22 unit tests; Linux visual baselines generated via the CI-matching Docker image (10 stories)

## Test plan
- [x] `vitest` geometry specs (22 passing)
- [x] eslint + typecheck clean
- [x] Visual snapshots regenerated and verified stable in Docker (Linux/CI renderer)
- [ ] Playwright Tests workflow green on CI

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added the `obc-gauge-valve` component for two-way and three-way valve visualization.
  * Supports flow values, setpoints, labels, units, tags, actuator icons, scale markers, compact and large layouts, and fill or tint styles.
  * Added regular, enhanced, medium, and off priority states.
  * Added Storybook examples for common valve configurations and display states.

* **Tests**
  * Added comprehensive coverage for valve geometry, scaling, clamping, coordinate conversion, radial paths, and edge cases.

Closes #1169

### CODE REVIEW

**Review by talpitoo — CHANGES_REQUESTED** [review]
@ludvigovrevik i think tint vs. fill should match https://openbridge-next-storybook.web.app/?path=/story/bars-and-graphs-bar-vertical--fill-mode-comparison (this is also in Figma), so in 'fill' mode we do not show those 'end rounded bars'.

Maybe you can reuse it from https://openbridge-next-storybook.web.app/?path=/story/instruments-gauge-radial--enhanced-filled and https://openbridge-next-storybook.web.app/?path=/story/instruments-gauge-radial--enhanced-bar ?

Screenshot from 2026-08-04 14-32-54

**Review by talpitoo — CHANGES_REQUESTED** [review]
## Sizing & reuse: adopt the shared radial-frame stack instead of a parallel width-only implementation

**Suggest rebasing on `develop` first.** `obc-gauge-radial-proportional` landed there on 2026-07-30 (#1067) — right at this branch's merge-base, so it isn't on `feat/gauge-valve` at all. It's the closest precedent for this component: the same Figma family (regular/enhanced/medium/off priority axes, center device icon, anchors from the same 512 canvas), built on the shared watch stack. Most of what's below can be lifted from it nearly verbatim, which wasn't visible when this branch was cut.

### 1. Sizing is width-driven only — out of contract with every sibling gauge

The chain in `gauge-valve.css` is `:host {display: block}` → `.root {width: 100%}` → `svg {width: 100%; height: auto}`: the face diameter is decided by container width alone and the container **height is never consulted** (the stories don't surface this because `widthDecorator` wraps them in `overflow: auto` boxes sized to fit). Every other radial instrument contain-fits both axes — `obc-watch` via `<svg width="100%" height="100%">` + `--scale = min(w,h)`, the proportional gauge via `aspect-ratio: 1` + `max-width/max-height: 100%` on a host that fills its container. Consequences:

1. **Fixed-height dashboard cells overflow.** In a 300×200 cell the valve renders a 300px face plus the fixed-px label stack and spills out, while any sibling gauge shrinks to fit — mixed rows can't line up.
2. **No equal-circumference story.** There's no `faceDiameter`, and the two variants have different face-to-box ratios (compact face ≈ 99.5% of width, 376/378 crop; large ≈ 73.4%, 376/512) — a large next to a compact at equal width shows a ~26% smaller ring, and matching a gauge-valve ring to any other radial instrument means hand-solving cell widths.
3. **Text/stroke scaling breaks the system contract.** The 0/50/100 scale labels are SVG text without the `calc(12px / var(--scale))` counter-scaling, so they grow/shrink with the face while the HTML readout/label stack stays fixed px — the two text systems drift apart as the gauge resizes. And `vector-effect="non-scaling-stroke"` is on `.track` only; the r188 `.outline` (1px) and `.tick-primary/-secondary` (2/1px) scale with the box, so at any size ≠ 512px the "1px" edges are unequal. (The cap pills are correct as-is — their 8/6px strokes *are* the pill geometry.)

### 2. Reuse / DRY: most of the rebuilt layer already exists

| Reimplemented in this PR | Existing shared code |
| --- | --- |
| `annularSectorPath` (`gauge-valve-geometry.ts`) | `svghelpers/roundedArch.ts` — rounded annular sector; the proportional gauge draws all its lanes with it |
| `polarToCartesian` / `arcPath` | `watch/secondary-lane.ts` `arcPath` + the center-origin helpers |
| Hand-rolled tick ladder + SVG scale labels (`renderScale`) | `tickmark.ts` — `tickmark()` with counter-scaled text, label offsets, `TickmarkType` sizes |
| Hardcoded `'0 0 512 512'` / `'67 67 378 378'` viewBoxes | `radial-frame.ts` — `computeRadialFrame()` + `measureContainerPx` + `applyPinnedHostSize` + `observeInnerBox` (contain-fit, label reserve, `faceDiameter`, resize handling) |
| `.setpoint-rotor` transform + CSS transition | `watch.ts` `renderSetpoint()` — shortest-path `cssSafeAngle`, dual-marker preview, and the `newSetpoint` confirm animation this PR's JSDoc lists as "not yet supported" (it's free on the watch path) |
| Sizing shell (`.root` / `.face-wrapper` / `.icon`) | `gauge-radial-proportional.css` — contain-fit root, `face-pinned`, `.container > .layer` stacking, `%`-of-face `.icon-anchor` (same center-icon-over-face concept) |

The tell that this was transcribed from Figma rather than mapped onto the core: **every radius here is the corresponding watch radius +4** (track 116–164 vs band 112–160; ticks 168→176/188 vs 164→172/184; outline 188 vs outer ring 184) — the known +4 offset of the Figma 6.1 canvas vs code watch radii, carried into code instead of normalized away.

What *is* already reused — `SetpointMixin`, the setpoint marker art, `renderInstrumentReadout`/`obc-readout`, shared enums, obi icons — is appreciated. And to be fair: `automation/` is outside the watch-instructions `applyTo` globs, the three fixed-angle port tracks are genuinely novel art, and today's watch `areas`/`needles` can't sit at the valve's band radii without new knobs. But the instructions' answer to that is to add configurable options to `watch.ts` (cf. the rate-of-turn "track-bar recipe": `barAreas` already take `innerRadius`/`outerRadius` overrides, and the cap pills are exactly the watch needle pill), not to fork the renderer. The large variant's scale/ticks/labels/setpoint is squarely watch/instrument-radial territory either way.

**Ask:** after rebasing, either (a) rebuild the face on `obc-watch` where the art maps and keep the genuinely novel port-track art as a matched-viewBox overlay using the proportional gauge's sizing shell, or (b) at minimum adopt `computeRadialFrame` + contain-fit + `faceDiameter` + the counter-scaling/`vector-effect` contract while keeping the drawing bespoke. If width-driven sizing is intentional for automation layouts, the JSDoc must say so explicitly — it's the opposite of every sibling gauge.

### 3. Related findings on develop (context, not caused by this PR)

Comparing against `watch-radial-instruments.instructions.md` surfaced pre-existing gaps that this PR would compound with a third parallel implementation:

- **The instructions doc doesn't know about `gauge-radial-proportional`**: it's missing from the `applyTo` globs, the architecture diagram, the component quick-reference table, and the `computeRadialFrame` consumers list. Whether automation-side radial gauges belong in those globs is the same decision this PR forces — worth settling in the doc.
- **`gauge-radial-proportional.renderNeedle()` duplicates `watch.renderNeedles()` verbatim** (x −4, y −160, 8×length pill, rx 4, `paint-order`) instead of passing `.needles` — apparently because its split-frame overlay must paint above watch output. The erase-and-redraw of the watch's end cuts in `renderSplitFrame()` is itself the kind of "new visual element in an individual instrument" the doc says should become a `watch.ts` option.
- **The interval→tickmark builder (`strongerTickmarkType` + `addTickmarksAtInterval` + min/max upserts) is copy-pasted** between `instrument-radial.ts` and `gauge-radial-proportional.ts`, and this PR's `renderScale()` is a third hand-rolled variant. One shared builder in `tickmark.ts` would serve all three.

File: packages/openbridge-webcomponents/__vis__/linux/__baselines__/automation/gauge-valve/gauge-valve.stories.ts/tint-vs-fill-auto.png
Comment (talpitoo) [inline packages/openbridge-webcomponents/__vis__/linux/__baselines__/automation/gauge-valve/gauge-valve.stories.ts/tint-vs-fill-auto.png:1]: @ludvigovrevik when fill=auto the bar is not rounded, please double-check in Figma. So this attached screenshot is off: 

Image

File: packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.stories.ts
Comment (talpitoo) [inline packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.stories.ts:1]: @ludvigovrevik nice to have: a 'manual' story to demonstrate `faceDiameter` like https://openbridge-next-storybook.web.app/?path=/story/instruments-gauge-radial--sizing-playground e.g. https://github.com/Ocean-Industries-Concept-Lab/openbridge-webcomponents/blob/6f5d92d66b5790ed34ab61f8da00c6fa36c44152/packages/openbridge-webcomponents/src/navigation-instruments/gauge-radial/gauge-radial.stories.ts#L517

File: packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.stories.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import '../../icons/icon-twoway-acuator-general-closed.js';
+import '../../icons/icon-threeway-acuator-generic-inleft-left-75.js';
+import {widthDecorator} from '../../storybook-util.js';
+
+type GaugeValveStoryArgs = Partial<ObcGaugeValve> & {
+  width?: number;
+  height?: number;
+};
+
+const meta = {
```
Comment (talpitoo) [inline packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.stories.ts:26]: @ludvigovrevik one more nice to have: width/height sliders instead of inputs and lockFaceDiameter so users can experiment. This is because the instruments is still not using the `watch.ts` so it is easier to spot if something is off in the future.

Image

File: packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.ts
Comment (talpitoo) [inline packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.ts:1]: @ludvigovrevik do you mind addressing the follow-up issues either in this PR or in a follow-up? :)

Those under "3. Related findings on develop (context, not caused by this PR)"

I am mostly concerned about not fully reusing `watch.ts` so at least we document it as an exception in `.github/instructions/watch-radial-instruments.instructions.md` or leave a TODO or a note that it is 73.8% according to the docs, rest is custom.

**Review by talpitoo — CHANGES_REQUESTED** [review]

File: packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.ts
Comment (ludvigovrevik) [reply packages/openbridge-webcomponents/src/automation/gauge-valve/gauge-valve.ts:1]: Good question — it's now documented as the exception in `watch-radial-instruments.instructions.md` (b7d0004c), but to spell out the reasoning:

**What the valve now shares with the watch stack** (since fbac1e28): `computeRadialFrame()` sizing (contain-fit, label reserve, `faceDiameter`), the watch coordinate space (center-origin, ring r184 — the +4 Figma-canvas offset normalized away), `roundedArch()`, the shared setpoint layer and `renderInstrumentReadout`. The scale ticks sit at literally the watch tickmark radii (164→172/184).

**Why the face isn't composed from `obc-watch`:** the valve's body is three *fixed-angle* sectors (centered 90°/180°/270°, each with an independent fill value) at band radii 112–160, with a 1px-bordered rounded track, a square-cut fill bar and cap pills per sector. watch's band/area model maps a single value axis around the dial at its own lane radii — rendering the valve tracks through it needs new watch options: band-radius overrides, static fixed-angle sectors with per-sector fill, and per-sector corner/border/pill styling. Since any `watch.ts` change has to be verified against every consumer (per the instructions), that extension would have roughly doubled this PR and moved the risk into the whole radial family — so I kept the genuinely novel track art bespoke per your option (b) and left the watch rebuild as the TODO in the instructions doc.

Happy to take the `watch.ts` extension + face rebuild as a follow-up PR — that path also picks up the `newSetpoint` confirm animation for free via the watch setpoint renderer.

**Review by talpitoo — APPROVED** [review]

**Comment by talpitoo** [comment]
### Branch updated with `develop`

I merged `origin/develop` into this branch (merge, not rebase, to preserve history):

- The only conflict was `.github/instructions/watch-radial-instruments.instructions.md` — on `develop` that file is now **generated** from `docs/agents/watch-radial-instruments.md` (`npm run agents:sync`). I ported this PR's additions (the gauge-valve glob, the `gauge-radial-proportional` quick-reference row, and the "Partial-reuse exception: `automation/gauge-valve`" section) into the source doc and regenerated.
- `develop` also added lifecycle-tag lint rules, so I added `@experimental` to the `ObcGaugeValve` JSDoc and the matching `'experimental'` story tag (`b0638e1`). Typecheck + eslint pass.

### Review notes vs `docs/agents/watch-radial-instruments.md`

The overall approach is understood and now documented as the sanctioned exception — these are **nice-to-haves / notes for the future refactor TODO**, not blockers:

1. **The distance to `obc-watch` is smaller than the exception note suggests.** The note says the 112–160 track band is something "watch.ts cannot currently render", but the mapping onto existing watch primitives is surprisingly close:
   - track annulus 112–160 = exactly `WatchCircleType.double`'s secondary-ring band (`RING2 160` / `RING3 112`), and `WatchBarArea`'s default radii;
   - the fill bars = `barAreas` at default radii;
   - the cap pills (8 wide, rounded, spanning 112–160) = the `WatchNeedle` pill shape byte-for-byte (`x=-4, y=-160, height 48, rx 4`);
   - the scale ticks reuse watch tick geometry numerically: `TICK_BASELINE_RADIUS 164 = 328/2`, secondary outer `172 = 344/2`, primary outer `184 = 368/2`.

   The genuine gaps are: per-sector band **tracks** detached from the outer ring with `roundRadius: 5` corner fillets (watch's `areas`/`roundBandCuts` model ties sector cuts to the full 184→inner silhouette), the fixed 60° top scale with labels at fixed SVG r 198 (vs watch's px-scaled label radius), and the center icon slot. Recording this mapping in the TODO would scope a future rebuild as roughly "add detached band-sector tracks + a corner-radius knob to `watch.ts`" rather than a rewrite.

2. **Tick radii are re-declared as local literals** (`164/172/184`). They currently match `tickmark.ts`'s primary/secondary radii exactly, but nothing keeps them in sync — if the shared tickmark geometry moves, the valve silently diverges. Nice-to-have: import the constants from `tickmark.ts` (or reuse `tickmark()` once the refactor above happens).

3. **Compact label stack duplication with #1085.** PR #1085 extends `obc-automation-button-readout-stack` with slot-projected row icons and the ghost-zero padding (dimmed `.pad` digits) that this PR hand-rolls in `renderPaddedValue()` + local CSS. Once both land, the valve's compact stack looks like it could become a `<obc-automation-button-readout-stack>` consumer (value rows, per-port icons, `tag` line) — one shared stack instead of two implementations of the same design.

4. **Setpoint confirm animation** (documented as unsupported here): when it's added, the `cssSafeAngle()` accumulated-angle + CSS-transition pattern from `watch.ts` / `docs/agents/setpoint.md` is the intended shared mechanism.

5. **Doc drift with #1085:** the quick-reference row this PR adds says `gauge-radial-proportional`, which #1085 renames to `gauge-proportional`. Whichever PR merges second needs a one-line touch-up in `docs/agents/watch-radial-instruments.md` (+ `agents:sync`).

On the plus side: `computeRadialFrame()` usage, the `labelsHidden` degradation, `calc(12px / var(--scale))` label sizing, `vector-effect="non-scaling-stroke"`, and the shared setpoint layer all follow the docs as written. 👍

*Merge + review done by Claude Code on behalf of @talpitoo.*

**Comment by talpitoo** [comment]
Merged `develop` in again after #1166.

Both conflicts were inside generated blocks — `AGENTS.md`'s routing table and `.github/copilot-instructions.md` are now output of `npm run agents:sync`. Took develop's side and regenerated from the merged `docs/agents/watch-radial-instruments.md`, so the gauge-valve glob, the quick-reference row and the partial-reuse exception section all survive. `.claude/rules/watch-radial-instruments.md` is a new mirror #1166 added and comes along with the regeneration.

Two things needed patching beyond the conflicts:

- `npm run lint:fix:property-docs` hoists the gauge-valve property docs into the class header, per #1166's convention.
- The exception section failed `format:check` — prettier now covers `docs/agents/*.md`. Reformatted and re-synced.

**Doc drift with #1085:** the quick-reference row this PR adds is named `gauge-radial-proportional`, which #1085 renames to `gauge-proportional`. It is correct against develop as it stands today, so nothing to do now — but whichever of the two lands second needs a one-line fix plus `agents:sync`.

`tsc`, `lint:slots`, `lint:agents`, `lint:eslint` (0 errors), `format:check` and the `rules` tests pass. None of the 10 baselines collide with anything develop touched.

Untouched from CodeRabbit, since it is PR content rather than merge fallout: `hasLabelStack` has `attribute: false` and no renderer binds it, so a Storybook control change to `false` does not reach the component.

**Comment by talpitoo** [comment]
Tracking issue: #1169 — created so the design intent, scope and follow-ups live outside the PR.

**Comment by talpitoo** [comment]
Superseded by #1174, which carries this branch by merge — when it lands, this PR is automatically marked **merged** (its tip is an ancestor). #1174 continues the work here: the bespoke face (tracks/bars/caps/scale/setpoint/label stack) becomes `obc-watch` inputs via two small per-area extensions (`WatchArea.roundRadius`, `WatchArea.outlined`), the caps are watch needles, the setpoint is the shared watch marker (confirm animation included), the compact stack is the shared `obc-automation-button-readout-stack`, and the designs' Direction/Scale rotation axis ships as `scalePosition`. The documented partial-reuse exception in `watch-radial-instruments.md` is deleted.

**Comment by talpitoo** [comment]
Re-implemented in https://github.com/Ocean-Industries-Concept-Lab/openbridge-webcomponents/pull/1174
